pkg/controllers/managementuser/rbac/project_handler.go

```diff
@@ -113,21 +124,25 @@ func (p *pLifecycle) ensureNamespacesAssigned(project *v3.Project) error {
 		if err = p.ensureSystemNamespaceAssigned(cluster, project); err != nil {
 			return err
 		}
+	default:
+		return nil
 	}
 
-	return nil
+	_, err = p.m.workload.Management.Management.Clusters("").Update(cluster)
+
+	return err
 }
 
 func (p *pLifecycle) ensureDefaultNamespaceAssigned(cluster *v3.Cluster, project *v3.Project) error {
-	_, err := v32.ClusterConditionDefaultNamespaceAssigned.DoUntilTrue(cluster.DeepCopy(), func() (runtime.Object, error) {
-		return nil, p.assignNamespacesToProject(project, projectpkg.Default)
+	_, err := v32.ClusterConditionDefaultNamespaceAssigned.DoUntilTrue(cluster, func() (runtime.Object, error) {
+		return cluster, p.assignNamespacesToProject(project, projectpkg.Default)
```

Comment:
You don't need to return the cluster object here if you aren't doing anything with it afterwards.

Reply:
Fixed, seems like I shouldn't have force pushed